File: numeric/euclidgcd/euclidgcd.nim
import os
import std/strutils
import std/strformat

proc euclidgcd(m_in: int, n_in: int): int =
    var
        r = 0
        m = m_in
        n = n_in
    while n != 0:
        r = m mod n
        m = n
        n = r
    return m

var
    m = 15
    n = 10

if paramCount() > 1:
    m = parseInt(paramStr(1))
    n = parseInt(paramStr(2))

let gcd = euclidgcd(m, n)

echo &"{m} {n}"
echo &"gcd: {gcd}"

File: numeric/euclidgcd/euclidgcd.pas
program euclidgcd;
    uses SysUtils;

    function gcd(m, n : integer) : integer;
        var r : integer;
    begin
        while n <> 0 do begin
            r := m mod n;
            m := n;
            n := r
        end;
        gcd := m
    end;

    var m, n, value : integer;
begin
    if ParamCount >= 2 then begin
        m := StrToInt(ParamStr(1));
        n := StrToInt(ParamStr(2))
    end
    else begin
        m := 15;
        n := 10;
    end;

    value := gcd(m, n);
    WriteLn(m, ' ', n);
    WriteLn('gcd: ', value)
end.
